### PR TITLE
Add differences controller

### PR DESCRIPTION
**Pull request by SpaskoKatsarski**
Add functionality for managing differences. Pagination included.

### CODE REVIEW

File: WarehouseManagement.Core/Services/DifferenceService.cs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    public async Task EditAsync(int id, DifferenceFormDto model, string userId)
+    {
+        if (!await ExistsByIdAsync(id))
+        {
+            throw new KeyNotFoundException(DifferenceWithIdNotFound);
+        }
+
+        var difference = await repository.GetByIdAsync<Difference>(id);
+        
+        difference!.ReceptionNumber = model.ReceptionNumber;
```
Comment (vexleet) [inline WarehouseManagement.Core/Services/DifferenceService.cs:62]: why the `!`?

File: WarehouseManagement.Core/Services/DifferenceService.cs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                Id = d.Id,
+                ReceptionNumber = d.ReceptionNumber,
+                InternalNumber = d.InternalNumber,
+                ActiveNumber = d.ActiveNumber,
+                Comment = d.Comment,
+                Count = d.Count,
+                Status = d.Status,
+                Type = d.Type.Name,
+                Zone = d.Zone.Name,
+                DeliverySystemNumber = d.Delivery.SystemNumber
```
Comment (vexleet) [inline WarehouseManagement.Core/Services/DifferenceService.cs:103]: what about AdminComment?

File: WarehouseManagement.Core/Services/DifferenceService.cs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
```
Comment (vexleet) [inline WarehouseManagement.Core/Services/DifferenceService.cs:130]: same here

File: WarehouseManagement.Core/Services/DifferenceService.cs [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+            Id = id,
+            ReceptionNumber = model!.ReceptionNumber,
+            InternalNumber = model.InternalNumber,
+            ActiveNumber = model.ActiveNumber,
+            Comment = model.Comment,
+            Count = model.Count,
+            Status = model.Status,
+            Type = model.Type.Name,
+            Zone = model.Zone.Name,
+            DeliverySystemNumber = model.Delivery.SystemNumber
```
Comment (vexleet) [inline WarehouseManagement.Core/Services/DifferenceService.cs:156]: same here with AdminComment

File: WarehouseManagement.Core/Services/DifferenceService.cs (hunk 2 above)
Comment (SpaskoKatsarski) [reply WarehouseManagement.Core/Services/DifferenceService.cs:103]: You sure you want to be included in the form? Wouldn't it be better to be passed on Finish Processing/No Differences?

File: WarehouseManagement.Core/Services/DifferenceService.cs (hunk 3 above)
Comment (SpaskoKatsarski) [reply WarehouseManagement.Core/Services/DifferenceService.cs]: Fixed

File: WarehouseManagement.Core/Services/DifferenceService.cs (hunk 4 above)
Comment (SpaskoKatsarski) [reply WarehouseManagement.Core/Services/DifferenceService.cs]: Fixed aswell

File: WarehouseManagement.Core/Services/DifferenceService.cs
Comment (vexleet) [inline WarehouseManagement.Core/Services/DifferenceService.cs:1]: lets add unit tests

Comment (SpaskoKatsarski) [reply WarehouseManagement.Core/Services/DifferenceService.cs:1]: Got you

**Review by vexleet — APPROVED** [review]

**Pull request merged**